### PR TITLE
Add #337 #338 AMI and add new customDeviceMapping config with new benchmark agent

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add #337 #338 AMI and add new customDeviceMapping config with new benchmark agent

### Issues Resolved
#337
#338 
Benchmark Host NewSpecs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
This PR add the AMI from the #337 #338 PR you approved.

And also add the new benchmark host for you to try out on C5.4xlarge 4executors on Jenkins.

It also fixed a but on the ubuntu agent as that one has different mounting point with customedevicemapping separate to config.

File: lib/ci-stack.ts [hunk 1]
```diff
@@ -161,6 +161,7 @@ export class CIStack extends Stack {
         agentNode.AL2_ARM64_DOCKER_HOST,
         agentNode.AL2023_ARM64_DOCKER_HOST,
         agentNode.AL2023_X64_DOCKER_HOST_BENCHMARK_TEST,
+        agentNode.AL2023_X64_DOCKER_HOST_BENCHMARK_TEST_NEW_SPEC,
```
Comment (rishabh6788) [inline lib/ci-stack.ts:164]: I think this is a good opportunity to remote "DOCKER" from the agent label as we run everything on the localhost and not inside a CI docker container. @peterzhuamazon

**Comment by rishabh6788** [comment]
LGTM, please take care of the name change for the benchmark host.

**Review by rishabh6788 — APPROVED** [review]

File: lib/ci-stack.ts (hunk 1 above)
Comment (peterzhuamazon) [reply lib/ci-stack.ts:164]: Will change, thanks!

**Pull request merged**